app/presenters/tree_node/menu/node.rb

```diff
@@ -0,0 +1,39 @@
+module TreeNode
+  module Menu
+    class Node < TreeNode::Node
+      set_attribute(:no_click, true)
+      set_attribute(:icon, "pficon pficon-folder-close")
+
+      set_attribute(:checkable) { @options[:editable] }
+
+      def self_and_child_states
+        child_states.unshift(selected)
+      end
+
+      private
+
+      def child_states
+        @states ||= children.flat_map(&:self_and_child_states)
+      end
+
+      def select_by_children_selected
+        return true        if all_children_fully_selected?
+        return 'undefined' if any_children_selected?
+
```

Comment:
I respectfully disagree, this is more readable. It is idiomatic Ruby to have these small predicate methods, especially to clean up what would otherwise be a nested `if` statement.